core/templates/pages/contributor-dashboard-page/services/contribution-and-review.service.spec.ts

```diff
@@ -13,96 +13,381 @@
-  var suggestion1;
-  var opportunityDict1;
-  var mockSuggestionsBackendObject;
-  var expectedSuggestionDict;
-  var onSuccess;
-
-  beforeEach(angular.mock.module('oppia'));
-  beforeEach(angular.mock.module('oppia', function($provide) {
-    var ugs = new UpgradedServices();
-    for (const [key, value] of Object.entries(ugs.getUpgradedServices())) {
-      $provide.value(key, value);
-    }
-  }));
-
-  beforeEach(angular.mock.inject(function($injector) {
-    ContributionAndReviewService = $injector.get(
-      'ContributionAndReviewService');
-    $httpBackend = $injector.get('$httpBackend');
-
-    suggestion1 = {
-      suggestion_id: 'suggestion_id_1',
-      target_id: 'skill_id_1',
-    };
-    opportunityDict1 = {
-      skill_id: 'skill_id_1',
-      skill_description: 'skill_description_1',
-    };
-    mockSuggestionsBackendObject = {
-      suggestions: [
-        suggestion1
-      ],
-      target_id_to_opportunity_dict: {
-        skill_id_1: opportunityDict1,
-      },
-    };
-    expectedSuggestionDict = {
-      suggestion: suggestion1,
-      details: opportunityDict1,
-    };
-  }));
+import { TestBed, fakeAsync, tick } from '@angular/core/testing';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { ContributionAndReviewService } from './contribution-and-review.service';
+import { UrlInterpolationService } from 'domain/utilities/url-interpolation.service';
+import { ContributionAndReviewBackendApiService }
+  from './contribution-and-review-backend-api.service';
+import { SuggestionBackendDict } from 'domain/suggestion/suggestion.model';
+
+describe('Contribution and review service', () => {
+  let cars: ContributionAndReviewService;
+  let carbas: ContributionAndReviewBackendApiService;
+
+  let suggestion1 = {
+    suggestion_id: 'suggestion_id_1',
+    target_id: 'skill_id_1',
+  } as SuggestionBackendDict;
+
+  let opportunityDict1 = {
+    skill_id: 'skill_id_1',
+    skill_description: 'skill_description_1',
+  };
+
+  let testSuggestionsBackendObject = {
+    suggestions: [
+      suggestion1
+    ],
+    target_id_to_opportunity_dict: {
+      skill_id_1: opportunityDict1,
+    },
+  };
+
+  let expectedSuggestionDict = {
+    suggestion: suggestion1,
+    details: testSuggestionsBackendObject
+      .target_id_to_opportunity_dict.skill_id_1
+  };
 
-  afterEach(function() {
-    $httpBackend.verifyNoOutstandingExpectation();
-    $httpBackend.verifyNoOutstandingRequest();
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [
+        UrlInterpolationService,
+        ContributionAndReviewBackendApiService
+      ]
+    });
+    cars = TestBed.inject(ContributionAndReviewService);
+    carbas = TestBed.inject(ContributionAndReviewBackendApiService);
   });
 
-  describe('getUserCreatedQuestionSuggestionsAsync', function() {
+  describe('getUserCreatedQuestionSuggestionsAsync', () => {
```

Comment:
This spec file needs a bit of restructuring and improvement. @aishwary023 @gp201 Can you guys please review this file?